services/core-web/src/components/mine/ExplosivesPermit/MineExplosivesPermitTable.tsx

```diff
@@ -94,177 +104,767 @@ const MineExplosivesPermitTable: FC<RouteComponentProps & MineExplosivesPermitTa
+                        {
+                          key: "view",
+                          label: "View",
+                          clickFunction: (event) =>
+                            handleOpenViewExplosivesPermitModal(event, record),
+                          icon: viewIcon,
+                        },
+                        {
+                          key: "process",
+                          label: "Process",
+                          clickFunction: (event) =>
+                            props.handleOpenExplosivesPermitDecisionModal(event, record),
+                          icon: editIcon,
+                        },
+                        {
+                          key: "edit",
+                          label: "Edit",
+                          clickFunction: (event) =>
+                            props.handleOpenAddExplosivesPermitModal(event, isPermitTab, record),
+                          icon: editIcon,
+                        },
+                      ]
+                    : []),
+                  {
+                    key: "0",
+                    label: "Edit Documents",
+                    clickFunction: (event) =>
+                      props.handleOpenAddExplosivesPermitModal(event, isPermitTab, record),
+                    icon: editIcon,
+                  },
+                ]
+              : [
+                  ...(!isProcessed
+                    ? [
+                        {
+                          key: "process",
+                          label: "Process",
+                          clickFunction: (event) =>
+                            props.handleOpenExplosivesPermitDecisionModal(event, record),
+                          icon: editIcon,
+                        },
+                        {
+                          key: "edit",
+                          label: "Edit",
+                          clickFunction: (event) =>
+                            props.handleOpenAddExplosivesPermitModal(event, isPermitTab, record),
+                          icon: editIcon,
+                        },
+                      ]
+                    : []),
+                  {
+                    key: "0",
+                    label: "Edit Documents",
+                    clickFunction: (event) =>
+                      props.handleOpenAddExplosivesPermitModal(event, isPermitTab, record),
+                    icon: editIcon,
+                  },
+                ];
+            const deleteAction: ITableAction = {
+              key: "delete",
+              label: "Delete",
+              clickFunction: (event) => {
+                deleteConfirmWrapper(
+                  `Explosives Storage & Use Permit${isPermitTab ? "" : " Application"}`,
+                  () => props.handleDeleteExplosivesPermit(event, record)
+                );
+              },
+              icon: <DeleteOutlined />,
+            };
+
+            const currentMenu = isApproved ? approvedMenu : menu;
```

Comment:
I really don't like that all of this is repeated.

Reply:
Oh gross.  I didn't even look at that.  Yeah, I'll clean that up in the morning.